elastic/beats
# libbeat/kibana: add ClientConfig.Headers

## What does this PR do?

Add configurable headers to include in every request sent to Kibana.

## Why is it important?

This is needed in some security applications, for authenticating the Beat. Specifically, this is needed to fix https://github.com/elastic/apm-server/issues/4065

## Checklist

~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

## How to test this PR locally

1. Run `nc -l 1234`
2. Configure filebeat with `setup.kibana.host: localhost:1234`, and a custom header in `setup.kibana.headers`
3. Run `filebeat setup`

Observe the custom header is logged by netcat:

<pre>
GET /api/status HTTP/1.1
Host: localhost:1234
User-Agent: Go-http-client/1.1
Accept: application/json
Content-Type: application/json
<b>Foo: bar</b>
Kbn-Xsrf: 1
Accept-Encoding: gzip
</pre>

## Related issues

https://github.com/elastic/apm-server/issues/4065

## Use cases

Introduces a means of specifying custom HTTP headers to add to all requests sent to Kibana. As above, needed in some security use cases.
